framework/src/mesh/StitchedMesh.C

```diff
@@ -0,0 +1,108 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "StitchedMesh.h"
+#include "Parser.h"
+#include "InputParameters.h"
+
+// libMesh includes
+#include "libmesh/mesh_modification.h"
+#include "libmesh/serial_mesh.h"
+#include "libmesh/exodusII_io.h"
+
+template<>
+InputParameters validParams<StitchedMesh>()
+{
+  InputParameters params = validParams<MooseMesh>();
+  params.addRequiredParam<std::vector<MeshFileName>>("files", "The name of the mesh files to read.  These mesh files will be 'stitched' into the current mesh in this order.");
+
+  params.addRequiredParam<std::vector<BoundaryName>>("stitch_boundaries", "Pairs of boundary names (one after the other) to stitch together for each step.");
```

Comment:
It would work fine - as long as both the first and second mesh have a sideset that is the same and is the one that you want to use to join to the third mesh.

Consider this:

```
Sidesets:
Mesh1: stuff, junk
Mesh2: otherstuff, junk
Mesh3: otherjunk

stitch(mesh1, mesh2, stuff, otherstuff) // adds mesh2 into mesh1 and merges the matching elements on stuff/otherstuff
stitch(mesh1, mesh3, junk, otherjunk) // adds mesh3 into "mesh1".  Since "mesh1" contains both mesh1 and mesh2 this actually merges all 3 together
```